python/kwiver/vital/tests/test_metadata.py

```diff
@@ -234,28 +234,29 @@ def check_initial_properties(self, inst, prop_info):
         nt.assert_equals(inst.name, prop_info.name)
         nt.assert_equals(inst.tag, prop_info.tag)
 
-        # A few tests on inst.data
-        nt.ok_(isinstance(inst.data, type(prop_info.data)))
-        if isinstance(prop_info.data, GeoPoint):
-            nt.assert_equals(inst.data.crs(), prop_info.data.crs())
-            if prop_info.data.is_empty():
-                nt.ok_(inst.data.is_empty())
-            else:
-                inst_loc = inst.data.location()
-                exp_loc = prop_info.data.location()
-                np.testing.assert_array_almost_equal(inst_loc, exp_loc)
-
-        elif isinstance(prop_info.data, GeoPolygon):
-            nt.assert_equals(inst.data.crs(), prop_info.data.crs())
-            if prop_info.data.is_empty():
-                nt.ok_(inst.data.is_empty())
-            else:
-                inst_verts = inst.data.polygon().get_vertices()
-                exp_verts = prop_info.data.polygon().get_vertices()
-                np.testing.assert_array_almost_equal(inst_verts, exp_verts)
+        if prop_info.data is not None:
```

Comment:
Does `inst.data()` not cause an error here? Seems like we're returning an instance of `unknown_t`, which is not bound

Reply:
Yes, that's the reason for adding the check. 

Would it be better perhaps to somehow bind the C++ `unknown_t` to Python's `None`, if possible? Or keep it this way? Or something else...

Reply:
I think treating `unknown_t` as `None` is reasonable on the Python side. I haven't used this class, and I wrote the bindings for it a while ago, however. @mwoehlke-kitware, does this sound reasonable to you?

Reply:
I've changed it to `None` on the python side.

Reply:
`type()` would also need to change. I'd like to get approval from either @mwoehlke-kitware or @russelldj before making the switch to `None`, however, as I've never used this class, just written the bindings

Reply:
I'm slightly lost here. Is `unknown_t` the class returned by a query which doesn't contain that metadata item? It seems like in the past we've compared the tag to the `VITAL_METADATA_UNKNOWN` tag. So if this were `None`, would that break?

Reply:
No, it shouldn't break. 
The previous behavior of `inst.data` was to return the integer value `0`. Now, it returns `None`, but the tag has not changed. 
If you think of the metadata like a `{ tag: data }` dictionary, the entry was `{ VITAL_METADATA_UNKNOWN: 0 }`; now it is `{ VITAL_METADATA_UNKNOWN: None }`. In my opinion, this makes more sense; an unknown tag doesn't have an integer value, it has no value (`None`).
On the C++ end, Python's built-in `None` type is replaced here with a dummy `unknown_t` class, the point of which is that, unlike the integer `0`, it cannot be accidentally interpreted as a legitimate value of any sort. Right now, if you do `metadata.find(SOME_TAG).as_double()` and the tag is not found, the `as_double()` will silently return `0.0`, since `int` can be casted to `double`. This is a problem because the program will continue on like the metadata field actually had the value `0.0`. With these new changes, `as_double()` will instead throw a runtime error, since `unknown_t` cannot be casted to `double`, alerting you to the bug.

Reply:
Thanks for the through response, it makes a lot more sense now. I agree that `None` is a very reasonable value.

Reply:
I think it would also be reasonable to change what `type()` returns as well.

Reply:
@tao558 What would you suggest? Currently it returns the fully-qualified name of the C++ type, which seems to be standard elsewhere in the file (e.g. `kwiver::vital::geo_point`)

Reply:
Right thats fine on the C++ side, but that type on the Python side is now `type(None)`, since we changed data to be None. We're still returning `typeid(unknown_t)` in the Python bindings even though we've changed the data to `None`

Reply:
To the string literal `"none"`, then? Or `"py::none"` or `"None"`?

Reply:
That gives me a `'py::type' has not been declared` error. I looked in the pybind11 repo and `git blame` says that `py::type` class was only added in the past year. I don't think Fletch's version of `pybind11` has that feature.